### ISSUE TITLE
UnionType not assignable to TypeForm

### ISSUE THREAD

**Issue by Andrew5057** [microsoft/pyright#11525]
Pyright currently rejects assigning a `UnionType` to a `TypeForm`. Contrived example:

```python
from typing_extensions import TypeForm

unparameterized: TypeForm = object | None
parameterized: TypeForm[object] = object | None
```

Output:

```none
main.py
 main.py:3:29 - error: Type "UnionType" is not assignable to declared type "TypeForm[Unknown]"
    "UnionType" is not assignable to "TypeForm[Unknown]" (reportAssignmentType)
  main.py:4:35 - error: Type "UnionType" is not assignable to declared type "TypeForm[object]"
    "UnionType" is not assignable to "TypeForm[object]" (reportAssignmentType)
2 errors, 0 warnings, 0 informations
```

This appears to be the root cause of #10793 and [this Stack Overflow question](https://stackoverflow.com/questions/79966645/pyright-fails-parsing-typeform-for-unions-with-embedded-containers). Tested with the pyright CLI (v1.1.411).

**Comment by LeanderK** [comment]
I think this works if you enable the experiments features. You can do this by adding a pyrightconfig.json with:

```json
{
    "enableExperimentalFeatures": true
}
```
